packages/measure-tools/README.md

```diff
@@ -0,0 +1,164 @@
+Below is an example of creating a vertical toolbar filled with the default measurement tools.
+
+ ```typescript
+const verticalToolbar =
+    <Toolbar
+    expandsTo={Direction.Right}
+    items={
+        <>
+        <ActionItemButton actionItem={MeasureToolDefinitions.measureDistanceToolCommand} />
+        <ActionItemButton actionItem={MeasureToolDefinitions.measureAreaToolCommand} />
+        <ActionItemButton actionItem={MeasureToolDefinitions.measureLocationToolCommand} />
+        <ActionItemButton actionItem={MeasureToolDefinitions.measureRadiusToolCommand} />
+        <ActionItemButton actionItem={MeasureToolDefinitions.measureAngleToolCommand} />
+        </>
+    }
+    />;
+```
+
+### Measurement Property Widget
+
+Most applications will have a property grid widget in the lower right corner of the 9-zone UI. Our measurements do not display properties in the element property grid, instead there is a separate widget that will
+display properties of any selected measurements. It normally is added as a separate tab to the same zone as the property grid. Below is example code from the 9-zone sample application with the measurement widget defined.
+
+In this code, the measurement widget is hidden until a measurement has been selected. With this code you can have measurements and elements selected and have two tabs to switch between properties of either.
+
+ ```typescript
+bottomRight={
+    <Zone defaultState={ZoneState.Open} allowsMerging={true}
+    widgets={[
+
+        <Widget id="Properties" control={PropertyGridWidget} defaultState={WidgetState.Closed} fillZone={true}
+        iconSpec="icon-properties-list" labelKey="MyApp:components.properties"
+        applicationData={{
+            iModelConnection: UiFramework.getIModelConnection(),
+            rulesetId: this._rulesetId,
+        }}
+        syncEventIds={[SyncUiEventId.SelectionSetChanged]}
+        stateFunc={PropertyGridWidget.generateWidgetStateFuncWithMemory()}
+        />,
+
+        <Widget id="Measurements" control={MeasurementWidget} defaultState={WidgetState.Hidden} fillZone={true}
+        iconSpec="icon-measure" labelKey="TestMeasurementApp:components.measurements"
+        applicationData={{
+            iModelConnection: UiFramework.getIModelConnection(),
+        }}
+        syncEventIds={[MeasurementSyncUiEventId.MeasurementSelectionSetChanged]}
+        stateFunc={MeasurementWidget.generateWidgetStateFuncWithMemory()}
+        />,
+    ]}
+    />
+}
+```
+
+![Measurement Widget](./docs/images/MTDocs-Widget.gif)
+
+# Advanced Topics
+
+## Action Toolbar
+
+Measurements support a right click context menu called the "Action Toolbar". By default it is not enabled, however it is completely customizable by the application. Similar to how tools are constructed by creating item definitions,
```

Comment:
Can you link to more docs on "Action Toolbar", is that on itwinjs.org anywhere?

Reply:
AFAIK, this is a custom feature implemented by civil not in core, but i do think this would be useful to be added there. @bsteinbk thoughts?

Reply:
We have the concept of a popup toolbar, but I no nothing about an "Action Toolbar". I also noticed a few uses of "document." to add and remove event listeners.  It is best to get the ownerDocument from an HTMLElement, so this feature is compatible with use in popout windows now available in UI 2.0.  If you continue to use PopupToolbar.tsx you should get the ownerDocument from this._wrapperRef.  See core popup toolbar info at https://github.com/imodeljs/imodeljs/blob/693b71c55a0525c42f6621a2262d50006b9abd8f/docs/learning/ui/abstract/UiAdmin.md#L99 and example use at https://github.com/imodeljs/imodeljs/blob/df437ba429b7fb1395eeae1c34a67baf9b9f7970/test-apps/ui-test-app/src/frontend/tools/AccuDrawPopupTools.tsx#L253 .

Reply:
I'll add an enhancement to add support for the popout windows, and address it in a later pr. thanks Bill :)